okhttp/src/main/java/okhttp3/internal/http/HttpEngine.java

```diff
@@ -499,6 +500,13 @@ public static boolean hasBody(Response response) {
    * content types the application is interested in.
    */
   private Request networkRequest(Request request) throws IOException {
+    if (!request.isHttps()) {
+      Authenticator authenticator = client.proxyAuthenticator();
```

Comment:
Uh oh, the current API won’t really work for pre-emptive auth unless we generate a fake response. Hmm.